yarpctest/bench.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package yarpctest
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/yarpc/api/peer"
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/peer/hostport"
+)
+
+// BenchmarkPeerListChooseFinish assesses the speed of choose + finish for each
+// of the YARPC peer choosers, with varying numbers of peers, and varying
+// latencies.
+// In order to remove exogenous behaviors from the simulation, each new request
+// may be finished any number of turns in the future.
+// One random request will be finished for every new peer chosen.
+// By increasing the size of the concurrency window, we can increase the
+// variance of pending requests on each individual peer.
+//
+// Size is the number of peers in the list.
+// For this benchmark, the number of peers is fixed.
+// All peers are connected.
+// Variance is the number of concurrent pending requests.
```

Comment:
I think that this would be clearer if we we renamed `variance` to `concurrentRequests`. It was